docs/src/content/en/docs/deployment/monorepo.mdx

```diff
@@ -0,0 +1,155 @@
+    <FileTree.Folder name="ui" />
+    <FileTree.Folder name="utils" />
+  </FileTree.Folder>
+  <FileTree.File name="package.json" />
+  <FileTree.File name="pnpm-workspace.yaml" />
+</FileTree>
+
+## Environment variables
+
+Environment variables like `OPENAI_API_KEY` should be stored in an `.env` file at the root of the Mastra application `(apps/api)`, for example:
+
+<FileTree>
+  <FileTree.Folder name="api" defaultOpen>
+    <FileTree.Folder name="src" defaultOpen>
+      <FileTree.Folder name="mastra" />
+    </FileTree.Folder>
+    <FileTree.File name=".env" />
+    <FileTree.File name="package.json" />
+    <FileTree.File name="tsconfig.json" />
+  </FileTree.Folder>
+</FileTree>
+
+## Build and dev configuration
+
+In a monorepo setup, all dependency packages used by Mastra must be built when the application is deployed. The root-level `build` script runs build commands for each workspace in the monorepo.
+
+```json filename="package.json"
+{
+  // ...
+  "scripts": {
+    "dev": "pnpm -r --parallel --filter api --filter web dev",
+    "build": "pnpm --filter ui build && pnpm --filter utils build && pnpm --filter web build && pnpm --filter api build"
+  }
+}
+```
+
+In a monorepo with both a Mastra application and a frontend, it's useful to define a root-level `dev` script that runs both services together. To run only the Mastra application, use a `pnpm` filtered command:
+
+```bash copy
+pnpm --filter api dev
+```
+
+## Compiler configuration (optional / advanced)
+
+Mastra supports both **CommonJS (CJS)** and **ECMAScript Modules (ESM)**, making it compatible with a variety of JavaScript environments. In a monorepo setup, you can choose the module system that best fits your project.
+
+### Module support
+
+Mastra provides dual builds for each module type:
+
+- **ESM** builds use `.js` or `.mjs` extensions
+- **CJS** builds use `.cjs` extensions
+
+This ensures Mastra can integrate into projects using either module format.
+
+### Build output
+
+When building Mastra in a monorepo, the build system (such as `tsc`, `esbuild`, or `Rollup`) should produce output that matches the selected module format. This ensures optimized builds for both ESM and CJS, maintaining compatibility with modern and legacy environments.
+
+- **ESM** for projects using `import`/`export` syntax
+- **CJS** for environments that use `require`/`module.exports`
+
+### Advanced configuration
+
+In your build pipeline, configure TypeScript to emit the correct module format. Here's an example `tsconfig.json` for ESM output in a monorepo setup:
+
+```json filename="packages/utils/tsconfig.json"
+{
+  "compilerOptions": {
+    "module": "ESNext",
+    "target": "ES2022",
+    "moduleResolution": "node",
+    "esModuleInterop": true,
+    "allowSyntheticDefaultImports": true,
+    "skipLibCheck": true,
+    "outDir": "dist",
+    "declaration": true,
+    "allowJs": true
+  },
+  "include": ["src/**/*"]
+}
+```
+> To target CommonJS, change to: `module: "CommonJS"`.
```

Comment:
I don't think we need this whole section because for `mastra build` it should just work (independent on how you build your packages) and for `mastra dev` the `transpilePackages` option should make it work.

So this would be a ``## Using `transpilePackages`` section instead.